public_html/wp-content/mu-plugins/blocks/assets/src/sessions/edit.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, pickBy, split } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+const apiFetch = wp.apiFetch;
+const { withSelect } = wp.data;
+const { Component, Fragment } = wp.element;
+const { addQueryArgs } = wp.url;
+
+/**
+ * Internal dependencies
+ */
+import SessionsBlockControls from './block-controls';
+import SessionsInspectorControls from './inspector-controls';
+import GridToolbar from '../shared/grid-layout/toolbar';
+
+const blockData = window.WordCampBlocks.sessions || {};
+
+const SESSIONS_ICON = 'list-view';
+const MAX_POSTS = 100;
+
+const ALL_POSTS_QUERY = {
+	orderby  : 'title',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+	_embed   : true,
+};
+
+const ALL_TERMS_QUERY = {
+	orderby  : 'name',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+};
+
+class SessionsEdit extends Component {
+	constructor( props ) {
+		super();
+
+		this.state = {
+			allSessionPosts      : null,
+			allSessionTracks     : null,
+			allSessionCategories : null,
+		};
+	}
+
+	componentWillMount() {
+		this.isStillMounted = true;
+
+		const allSessionPosts = apiFetch( {
+			path: addQueryArgs( `/wp/v2/sessions`, ALL_POSTS_QUERY ),
+		} );
+		const allSessionTracks = apiFetch( {
+			path: addQueryArgs( `/wp/v2/session_track`, ALL_TERMS_QUERY ),
+		} );
+		const allSessionCategories = apiFetch( {
+			path: addQueryArgs( `/wp/v2/session_category`, ALL_TERMS_QUERY ),
+		} );
+
+		if ( this.isStillMounted ) {
+			this.setState( {
+				allSessionPosts      : allSessionPosts, // Promise
+				allSessionTracks     : allSessionTracks, // Promise
+				allSessionCategories : allSessionCategories, // Promise
+			} );
+		}
+	}
+
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
+	render() {
+		const { mode } = this.props.attributes;
+
+		return (
+			<Fragment>
+				<SessionsBlockControls
+					icon={ SESSIONS_ICON }
+					{ ...this.props }
+					{ ...this.state }
+				/>
+				{ mode &&
+				<Fragment>
+					<SessionsInspectorControls { ...this.props } />
+					<GridToolbar { ...this.props } />
+				</Fragment>
+				}
+			</Fragment>
+		);
+	}
+}
+
+const sessionsSelect = ( select, props ) => {
+	const { mode, item_ids, sort } = props.attributes;
+	const { getEntityRecords } = select( 'core' );
+
+	const args = {
+		per_page           : MAX_POSTS, // -1 is not allowed for per_page.
+		_embed             : true,
+		context            : 'view',
+		_wcpt_session_type : 'session',
+	};
+
+	if ( 'session_time' !== sort ) {
+		const [ orderby, order ] = split( sort, '_', 2 );
+		args.orderby = orderby;
+		args.order = order;
```

Comment:
`[SHOULD]` I mentioned this before in a regular comment, but I should have put that in a code comment so that it can be tracked as unresolved.

We should run ESLint and PHPCS before merging any commits from this PR.

Reply:
We should probably run it twice, actually. Once for new code, and once for existing code. That way we can keep the two commits separate during the merge to SVN. The commit on new code could be squashed into the others, but the commit on existing code should be kept separate, so that it doesn't add a ton of diff noise.

Reply:
Added PR for this https://github.com/WordPress/wordcamp.org/pull/42